### ISSUE TITLE
Huge Inconsistency in inference outputs between Triton with tensorrtllm_backend and tensorrtllm

### ISSUE THREAD

**Issue by CaesarWWK** [triton-inference-server/tensorrtllm_backend#138]
In my inference setup, I used build.sh to build my llama-based model (without any quantization and tp=8).

In tensorrtllm inference, I used run.py provided by tensorrtllm repo [https://github.com/NVIDIA/TensorRT-LLM/blob/release/0.5.0/examples/llama/run.py](url)

In the Triton with tensorrtllm_backend, I copied all_models/inflight_batcher_llm and turned off the inflight-batch.

I keep the same inference args (repetition penalty and num_beam)

In the tensorrtllm inference, the model output is usual as expected. But in the Triton inference, the output contains severe repetition problem and other errors.  My colleagues also report the similar issue (inconsistency in huggingface inference and triton inference ) on GPT based models.

I thought the problem may come from the C++ runtime in Triton's tensorrtllm backend. 
I will try to switch to Python runtime (switch to all_models/gpt template I think?) to see whether the problem still exists.

**Comment by CaesarWWK** [comment]
BTW, When I inference on the huggingface framework using Transformers==4.31.*, I got the exactly same error pattern (severe repetition). But when I update Transformers to 4.34.1, the problem got fixed.

But the requirement for tensorrt-llm 0.5.0 is Transformers==4.31.*, not 4.34.1 

So I guess the C++ backend in triton used some code in old version of Transformers, therefore they got the same error pattern?

**Comment by CaesarWWK** [comment]
UPDATE: I resolved the issue by switching to python backend. The step-by-step instruction is as below:

1. Install tensorrt-llm on your triton docker image, you can follow the installation instruction from the tensorrt-llm repo.  CAUTION: 23.08 triton image used in Nvidia's tutorial has error in triton api inference, I used 23.10 instead.
2. Modify the config in all_models/your_model/tensorrt-llm. Change backend form "tensorrtllm" to "python", set decoupled to false: model_transaction_policy {
  decoupled: false
}
3. Modify the config in all_models/your_model/ensemble, ensure engine_dir exists and put the path of your build tensorrt engine in it
4. Move the model.py from all_models/gpt to all_models/tensorrt-llm/1/  
5. set stop_id and pad_id in your inference requests

**Comment by byshiue** [comment]
Thank you for the update. Do you mean that you can get correct result on python backend, so you suspect the reason is caused by c++ backend?

**Comment by CaesarWWK** [comment]
> Thank you for the update. Do you mean that you can get correct result on python backend, so you suspect the reason is caused by c++ backend?

Yes.  Do you think the version of Transformers has something to do with the bug? (old version of Transformer has the same issue, when I switch to newer version it got fixed without any change to the code, and the requirement for triton is based on the old transformer )

**Comment by byshiue** [comment]
Is the issue of c++ backend also fixed on newer Transformer? 

Might you help to print the encoded ids after pre-processing for old and new Transformer?
